sdks/go/pkg/beam/pardo.go

```diff
@@ -116,6 +116,22 @@ func TryParDo(s Scope, dofn any, col PCollection, opts ...Option) ([]PCollection
 		}
 	}
 
+	wc := inWfn.Coder()
+	pipelineTimers := fn.PipelineTimers()
+	if len(pipelineTimers) > 0 {
+		c, err := inferCoder(typex.New(reflect.TypeOf(col.Type())))
+		if err != nil {
+			return nil, addParDoCtx(errors.New("error infering coder from col"), s)
+		}
+		edge.TimerCoders = make(map[string]*coder.Coder)
```

Comment:
1. If the coder is always going to be the Key of the input PCollection KV type, do we need to store a different coder for every timer family, since they're identical?

This means we could also avoid infering the coder multiple times, since this could be done in the graphx/translate.go step instead, and we can probably avoid filtering down to the Key type a few times in the exec side. (I should be able to comment on those directly to help track them down...)

Reply:
done